x-pack/plugins/security_solution/common/detection_engine/types.ts

```diff
@@ -18,7 +18,7 @@ export type RuleAlertAction = Omit<RuleAction, 'actionTypeId'> & {
  * "| { [property: string]: SearchTypes }" to designate you can can sub-objects
  * or sub-sub-objects, etc...
  */
-export type SearchTypes = string | number | boolean | object | SearchTypes[] | undefined;
+export type SearchTypes = string | number | boolean | object | SearchTypes[] | undefined | null;
```

Comment:
making timelineId and timelineTitle nullable requires changes in this typing as well